services/s3/src/main/java/software/amazon/awssdk/services/s3/model/GetUrlRequest.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.s3.model;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.core.RequestOverrideConfiguration;
+import software.amazon.awssdk.core.SdkField;
+import software.amazon.awssdk.core.SdkPojo;
+import software.amazon.awssdk.core.SdkRequest;
+import software.amazon.awssdk.core.protocol.MarshallLocation;
+import software.amazon.awssdk.core.protocol.MarshallingType;
+import software.amazon.awssdk.core.traits.LocationTrait;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.utils.builder.CopyableBuilder;
+import software.amazon.awssdk.utils.builder.ToCopyableBuilder;
+
+/**
+ * Request to generate a URL representing an object in Amazon S3.
+ *
+ * If the object identified by the given bucket and key has public read permissions,
+ * then this URL can be directly accessed to retrieve the object's data.
+ */
+@SdkPublicApi
+public final class GetUrlRequest extends SdkRequest implements ToCopyableBuilder<GetUrlRequest.Builder, GetUrlRequest> {
+    private static final SdkField<String> BUCKET_FIELD = SdkField
```

Comment:
Need these fields as this class extend `SdkRequest` which is required to create `ModifyHttpRequest` instance. Also `BUCKET_FIELD` is used by `EndpointAddressInterceptor`

Reply:
I don't really agree with this extending `SdkRequest`. Ignoring implementation details, this allows the customer to override api call timeouts, headers, query strings, etc. That doesn't make sense.

Reply:
Refactored